test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs

```diff
@@ -728,20 +728,21 @@ public class ModelBindingTest
 
             //Assert
             var json = JsonConvert.DeserializeObject<Dictionary<string, string>>(response);
+
             // 8 is the value of MaxModelValidationErrors for the application being tested.
             Assert.Equal(8, json.Count);
-            Assert.Equal("The Field1 field is required.", json["Field1.Field1"]);
-            Assert.Equal("The Field2 field is required.", json["Field1.Field2"]);
-            Assert.Equal("The Field3 field is required.", json["Field1.Field3"]);
-            Assert.Equal("The Field1 field is required.", json["Field2.Field1"]);
-            Assert.Equal("The Field2 field is required.", json["Field2.Field2"]);
-            Assert.Equal("The Field3 field is required.", json["Field2.Field3"]);
-            Assert.Equal("The Field1 field is required.", json["Field3.Field1"]);
+            Assert.Equal("The Field1 field is required.", json["Field1"]);
+            Assert.Equal("The Field2 field is required.", json["Field2"]);
+            Assert.Equal("The Field3 field is required.", json["Field3"]);
+            Assert.Equal("The Field4 field is required.", json["Field4"]);
+            Assert.Equal("The Field5 field is required.", json["Field5"]);
+            Assert.Equal("The Field6 field is required.", json["Field6"]);
+            Assert.Equal("The Field7 field is required.", json["Field7"]);
```

Comment:
fyi this change should not have been necessary.  also indicates a bug because the query provides values for the top-level properties.

the broken `MutableObjectModelBinder` must have been hiding problems in another binder. looking for the binder that returns `null` or a result with `!IsModelSet` when prefixes match in the value providers...